.github/workflows/code.yml

```diff
@@ -0,0 +1,213 @@
+      # next 5 lines to restore the original behaviour
+      - master
+      - main
+    tags:
+      - "*"
+  pull_request:
+  schedule:
+    # Run every Monday at 8am to check latest versions of dependencies
+    - cron: "0 8 * * MON"
+
+jobs:
+  lint:
+    runs-on: "ubuntu-latest"
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+
+      - name: lint
+        run: |
+          pip install --user .[dev]
+          tox -e pre-commit,mypy
+
+  wheel:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ["ubuntu-latest"]
+        python: ["3.8"]
+
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Create Sdist and Wheel
+        # Set SOURCE_DATE_EPOCH from git commit for reproducible build
+        # https://reproducible-builds.org/
+        # Set group writable and umask to do the same to match inside DLS
+        run: |
+          chmod -R g+w .
+          umask 0002
+          SOURCE_DATE_EPOCH=$(git log -1 --pretty=%ct) pipx run build --sdist --wheel
+
+      - name: Upload Wheel and Sdist as artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: dist
+          path: dist
+
+      - name: Install minimum python version
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python }}
+
+      - name: Install wheel in a venv and check cli works
+        # ${GITHUB_REPOSITORY##*/} is the repo name without org
+        # Replace this with the cli command if different to the repo name
+        run: pipx run --python $(which python${{ matrix.python }}) --spec dist/*.whl ${GITHUB_REPOSITORY##*/} --version
+
+  test:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ["ubuntu-latest"] # can add windows-latest, macos-latest
+        python: ["3.8", "3.9", "3.10"]
+        lock: [false]
+
+        include:
+          # Add an extra Python3.10 runner to use the lockfile
+          - os: "ubuntu-latest"
+            python: "3.10"
+            lock: true
+
+    runs-on: ${{ matrix.os }}
+    env:
+      # https://github.com/pytest-dev/pytest/issues/2042
+      PY_IGNORE_IMPORTMISMATCH: "1"
+      # enable QT tests with no X Display
+      QT_QPA_PLATFORM: "offscreen"
+
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+
+      - name: setup python ${{ matrix.python }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python }}
+
+      - name: create requirements.txt
+        if: ${{ matrix.lock == true }}
+        run: |
+          touch requirements.txt
+          pip install -r requirements.txt -e .
+          pip freeze --exclude-editable > requirements.txt
+
+      - name: run tests
+        run: |
+          if [ '${{ matrix.lock == true }}' ]; then lock='-r requirements_dev.txt'; fi
+          touch requirements_dev.txt
+          pip install ${lock} -e .[dev]
```

Comment:
```suggestion
          touch requirements_dev.txt
          pip install -r requirements_dev.txt -e .[dev]
```

Reply:
But this will always lock the developer install if there is a requirements_dev.txt. We want the test matrix to be mostly unlocked.